apps/webapp/app/hooks/useJobs.tsx

```diff
@@ -0,0 +1,26 @@
+import { UseDataFunctionReturn } from "remix-typedjson";
+import invariant from "tiny-invariant";
+import type { loader } from "~/routes/_app.orgs.$organizationSlug.projects.$projectParam.jobs.$jobParam/route";
+import { RouteMatch } from "@remix-run/react";
+import { useTypedMatchesData } from "./useTypedMatchData";
+
+export type ProjectJob = UseDataFunctionReturn<
+  typeof loader
+>["projectJobs"][number];
+
+export const jobsMatchId =
+  "routes/_app.orgs.$organizationSlug.projects.$projectParam.jobs.$jobParam";
```

Comment:
This one is a little tricky. I think you're correct that `routes/_app.orgs.$organizationSlug.projects.$projectParam.jobs._index` is the more accurate id for that page and job breadcrumb (this was my initial thought as well).

The problem is that we _also_ need the project jobs for the `useJob` hook. For example on the [job trigger page](https://github.com/triggerdotdev/trigger.dev/blob/main/apps/webapp/app/routes/_app.orgs.%24organizationSlug.projects.%24projectParam.jobs.%24jobParam.runs.%24runParam.trigger/route.tsx#L28). This means that we have to include the project jobs on the route `routes/_app.orgs.$organizationSlug.projects.$projectParam.jobs.$jobParam` because the trigger page _won't_ match `routes/_app.orgs.$organizationSlug.projects.$projectParam.jobs.$jobParam._index`.

If we make the suggested change for `useJobs` and make it `routes/_app.orgs.$organizationSlug.projects.$projectParam.jobs.$jobParam._index` then we would need to return the project jobs from both the `routes/_app.orgs.$organizationSlug.projects.$projectParam.jobs.$jobParam` and `routes/_app.orgs.$organizationSlug.projects.$projectParam.jobs.$jobParam._index` routes to account for both situations.

While that should work fine, I think the implication is that if you're on the `routes/_app.orgs.$organizationSlug.projects.$projectParam.jobs.$jobParam._index` page then we would make the same exact call from both loaders which seems wasteful. This is why I ended up using this id instead.

If you prefer we just make the call from both places, then I can update this id and then also update the associated loader to return the project jobs.

Reply:
Okay, that makes sense, and thanks for the clarification!